### PR TITLE
FoxyStripeSiteConfig - requireDefaultRecords

### LINKED ISSUE #217: SiteConfig.StoreKey not being generated in a fresh install
It works if I install foxystripe before running the first dev/build. It doesn't work if I dev/build the base install, and install foxystripe after.

I think the problem is populateDefaults only works when something is created. If foxystripe isn't installed initially, the SiteConfig is already created so setStoreKey is never triggered.

Is there another "on module install" trigger we could use instead?

### PR DESCRIPTION
**Pull request by jsirish**
fixes #217

generate StoreKey if empty or NULL

updated StoreSettingsTest to use requireDefaultRecords

### CODE REVIEW

**Comment by jsirish** [comment]
sorry for all the formatting changes, PHPStorm cleaned up when I copy/pasted from my other repo

**Comment by jsirish** [comment]
sorry for all the formatting updates, PHPStore cleaned up when I copy/pasted from my other repo

File: code/extensions/FoxyStripeSiteConfig.php [hunk 1]
```diff
@@ -141,4 +130,23 @@ private static function getDataFeedLink() {
...
 
+    // generate key on install
+    public function requireDefaultRecords() {
+
+        parent::requireDefaultRecords();
+
+        $siteConfig = SiteConfig::current_site_config();
+
+        if(!$siteConfig->StoreKey) {
+            debug::show('no key');
```
Comment (muskie9) [inline code/extensions/FoxyStripeSiteConfig.php]: This should be removed.

**Comment by muskie9** [comment]
It's almost like the .editorconfig didn't get picked up which is weird... should be the same one from the SS installer.

File: tests/StoreSettingsTest.php [hunk 2]
```diff
@@ -2,29 +2,31 @@
...
+        $this->assertEquals(substr($siteConf->StoreKey, 0, 6), $pref);
+    }
 
-		$this->assertTrue($siteConf->StoreName == 'foxystripe');
-	}
+    function testStoreName(){
+        $siteConf = SiteConfig::current_site_config();
 
-}
+        $this->assertTrue($siteConf->StoreName == 'foxystripe');
```
Comment (muskie9) [inline tests/StoreSettingsTest.php]: This can become `$this->assertEquals($siteConf->StoreName, 'foxy stripe');`

**Pull request merged**